sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -761,8 +804,10 @@ case class HashAggregateExec(
     val bufferTerm = ctx.freshName("aggBuffer")
     val outputFunc = generateResultFunction(ctx)
 
+    val limitNotReachedCondition = limitNotReachedCond
```

Comment:
Adding the limit early termination for first level map as well. This is needed to fix test failure `SQLMetricsSuite.SPARK-25497: LIMIT within whole stage codegen should not consume all the inputs` in https://github.com/c21/spark/runs/2386397792?check_suite_focus=true. And this is good to have anyway.